XiEditor/Search.swift

```diff
@@ -163,14 +163,26 @@ class FindViewController: NSViewController, NSSearchFieldDelegate, NSControlText
 
 extension EditViewController {
     func openFind(replaceHidden: Bool) {
-        if findViewController.view.isHidden {
-            findViewController.view.isHidden = false
-            document.sendRpcAsync("highlight_find", params: ["visible": true])
+        let replaceHiddenChanged = findViewController.replacePanel.isHidden != replaceHidden
+
+        if !findViewController.view.isHidden && replaceHiddenChanged {
+            let offset = findViewController.view.fittingSize.height
+            let origin = scrollView.contentView.visibleRect.origin
+            scrollView.contentView.scroll(to: NSMakePoint(origin.x ,origin.y + offset))
         }
 
         findViewController.replacePanel.isHidden = replaceHidden
         let offset = findViewController.view.fittingSize.height
```

Comment:
we declare `offset` and `origin` twice. Maybe we can just declare them once at the top of the fn?

Reply:
Although we declare them twice, they are declared in two different functions, `openFind()` and `closeFind()`. If we want to share them between two functions, then we should extract these variables to `EditViewController` class.

Am I understand correctly?

Reply:
Done

Reply:
`findViewController.replacePanel.isHidden = replaceHidden` may affect the value of `findViewController.view.fittingSize.height` and `scrollView.contentView.visibleRect.origin`. If replace bar is gone, then the height of the the entire find bar will be changed; similarly, the origin will be different too. 

As a result, we have to assign these values twice.

Reply:
oookay, thank you for the explanation, I was confused. Sorry!

Let's go back to how you had it before this change, and then we can get that merged.

This logic will also need to be updated when #229 is merged.

Reply:
Ah sorry I missed this comment.

All that I meant was that my earlier advice to declare `offset` and `origin` once, at the top, was because I did not understand that those values changed at various times in the function.

Now that I understand, I think that you were correct the first time, and that my advice made the code worse. 🙇 

This is not a big problem, though, and I will merge the current version in a few days.

Sorry for the confusion, and thank you for the patch! 🙏